### PR TITLE
Add camera position extension functions for camera position 

### PR DESCRIPTION
**Pull request by tobrun**
Adds extension functions for move/ease and animate camera. 
This avoids the need to make a complete CameraPosition from a Builder. 

**update**: This PR now also updates the module testsuite to use mockk, junit5 and AssertJ
Camera position tests are currently blocked by https://github.com/mapbox/mapbox-gl-native/issues/13075.

### CODE REVIEW

**Comment by tobrun** [comment]
Thanks for 👀 @LukasPaczos , learned something new! 
Should be fixed with https://github.com/mapbox/mapbox-plugins-android/pull/715/commits/1e60381793383c3d33fd79253ffde39ef40d0295.

File: ktx-mapbox-maps/src/main/java/com/mapbox/mapboxsdk/plugins/maps/Camera.kt [hunk 1]
```diff
@@ -0,0 +1,339 @@
...
+import com.mapbox.mapboxsdk.maps.MapboxMap
+
+//
+// Move camera
+//
+
+/**
+ * Repositions the camera to a latitude-longitude pair.
+ * The move is instantaneous, and a subsequent getCameraPosition() will reflect the new position.
+ * See CameraUpdateFactory for a set of updates.
```
Comment (LukasPaczos) [inline ktx-mapbox-maps/src/main/java/com/mapbox/mapboxsdk/plugins/maps/Camera.kt]: Maybe we could link to `CameraUpdateFactory` here?

File: ktx-mapbox-maps/src/main/java/com/mapbox/mapboxsdk/plugins/maps/Camera.kt [hunk 2]
```diff
@@ -0,0 +1,339 @@
...
+import com.mapbox.mapboxsdk.geometry.LatLng
+import com.mapbox.mapboxsdk.maps.MapboxMap
+
+//
+// Move camera
+//
+
+/**
+ * Repositions the camera to a latitude-longitude pair.
+ * The move is instantaneous, and a subsequent getCameraPosition() will reflect the new position.
```
Comment (LukasPaczos) [inline ktx-mapbox-maps/src/main/java/com/mapbox/mapboxsdk/plugins/maps/Camera.kt]: Maybe we could link to `getCameraPosition()` here?

File: ktx-mapbox-maps/src/main/java/com/mapbox/mapboxsdk/plugins/maps/Projection.kt [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+ *
+ * @param location A LatLng on the map to convert to a screen location.
+ * @return A Point representing the screen location in screen pixels.
+ */
+fun MapboxMap.toScreenLocation(latLng: LatLng): PointF {
+    return this.projection.toScreenLocation(latLng)
+}
+
+/**
+ *
```
Comment (LukasPaczos) [inline ktx-mapbox-maps/src/main/java/com/mapbox/mapboxsdk/plugins/maps/Projection.kt]: Missing javadoc.

File: ktx-mapbox-maps/src/test/java/com/mapbox/mapboxsdk/maps/ProjectionExtensionFunctionTest.kt [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+        every { projection.toScreenLocation(any()) } answers { PointF() }
+        every { projection.fromScreenLocation(any()) } answers { LatLng() }
+        mapboxMap = MapboxMap(null, null, null, projection, null, annotationManager, null)
+    }
+
+    @Test
+    fun `to screen location`() {
+        val latLng = LatLng()
+        mapboxMap.toScreenLocation(latLng)
+        verify(exactly = 1) { projection.toScreenLocation(latLng) }
```
Comment (LukasPaczos) [inline ktx-mapbox-maps/src/test/java/com/mapbox/mapboxsdk/maps/ProjectionExtensionFunctionTest.kt]: Can we also verify that the returned value is the one provided by the `projection.toScreenLocation(latLng)`?

File: ktx-mapbox-maps/src/main/java/com/mapbox/mapboxsdk/plugins/maps/Query.kt [hunk 5]
```diff
@@ -27,9 +26,7 @@ inline fun MapboxMap.queryRenderedFeatures(latLng: LatLng,
  * @param layerIds    optionally - only query these layers
  * @return the list of feature
  */
-inline fun MapboxMap.queryRenderedFeatures(latLng: LatLng,
-                                           filter: Expression?,
-                                           vararg layerIds: String): List<Feature> {
+fun MapboxMap.queryRenderedFeatures(latLng: LatLng, filter: Expression?, vararg layerIds: String): List<Feature> {
```
Comment (LukasPaczos) [inline ktx-mapbox-maps/src/main/java/com/mapbox/mapboxsdk/plugins/maps/Query.kt]: What if we make `filter` and `layerIds` default to `null`? This should suffice for both of methods and add an option of passing only `latLng`.

File: ktx-mapbox-maps/src/test/java/com/mapbox/mapboxsdk/maps/QueryExtensionFunctionTest.kt [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+        every { nativeMapView.queryRenderedFeatures(any<PointF>(), any(), any()) } answers { listOf() }
+        every { nativeMapView.queryRenderedFeatures(any<RectF>(), any(), any()) } answers { listOf() }
+        mapboxMap = MapboxMap(nativeMapView, null, null, projection, null, annotationManager, null)
+    }
+
+    @Test
+    fun `query with LatLng`() {
+        val latLng = LatLng()
+        mapboxMap.queryRenderedFeatures(latLng, "test")
+        verify(exactly = 1) { nativeMapView.queryRenderedFeatures(pointF, arrayOf("test"), null) }
```
Comment (LukasPaczos) [inline ktx-mapbox-maps/src/test/java/com/mapbox/mapboxsdk/maps/QueryExtensionFunctionTest.kt]: Can we also check that the extension function is returning the result of `nativeMapView.queryRenderedFeatures`?

**Pull request closed without merging**